src/tfr_control/src/test_client.cpp

```diff
@@ -0,0 +1,97 @@
+/****************************************************************************************
+ * File:    example_action_client.cpp
+ * Node:    example_client
+ * 
+ * Purpose: This is a example of what an ActionServer client node might look like.
+ *          It includes functions likely to be used in our production code for
+ *          ActionServer clients, but does not include every function available
+ *          to a client. If you are adapting this code for your system, please
+ *          read the ROS API documentation on actionlib::SimpleActionClient<> at 
+ *          http://docs.ros.org/api/actionlib/html/classactionlib_1_1SimpleActionClient.html
+ * 
+ *          This file includes <tfr_msgs/ExampleAction.h>, which is one of seven headers
+ *          built by catkin from `tfr_msgs/action/Example.action`:
+ *              devel/include/tfr_msgs/ExampleAction.h
+ *              devel/include/tfr_msgs/ExampleActionFeedback.h
+ *              devel/include/tfr_msgs/ExampleActionGoal.h
+ *              devel/include/tfr_msgs/ExampleActionResult.h
+ *              devel/include/tfr_msgs/ExampleFeedback.h
+ *              devel/include/tfr_msgs/ExampleGoal.h
+ *              devel/include/tfr_msgs/ExampleResult.h
+ * 
+ ***************************************************************************************/
+
+#include <tfr_msgs/ArmMoveAction.h> // Note: "Action" is appended
+#include <actionlib/client/simple_action_client.h>
+#include <std_msgs/Float64.h>
+#include <vector>
+#include <map>
+
+typedef actionlib::SimpleActionClient<tfr_msgs::ArmMoveAction> Client;
+
+// Called once when the goal completes
+void finished(const actionlib::SimpleClientGoalState& state, const tfr_msgs::ArmMoveResultConstPtr& result)
+{
+    // If this goal didn't succeed (there's multiple fail states in the enum so
+    // it's easier to check for this)
+    if (state.state_ != actionlib::SimpleClientGoalState::StateEnum::SUCCEEDED) {
+        ROS_WARN("Error planning/executing motion.");
+        return;
+    }
+
+    ROS_INFO("Planning and execution succeeded.");
+}
+
+int main(int argc, char** argv)
+{
+    // The list of positions to travel to, in order
+    std::vector<std::string> position_names = { "ready", "dig", "scoop", "out", "dump", "release", "invalid" };
+
+    // Basic ROS setup
+    ros::init(argc, argv, "example_client");
+    ros::NodeHandle n;
+
+    // Needed to spin while we're waiting for a blocking call later (waiting for
+    // server finished response)
+    ros::AsyncSpinner spin(1);
```

Comment:
Does everything that interacts with the moveit library need to be spun asynchronously?

Reply:
I don't think everything does (might need some more testing on that), but it seems that instantiating MoveIt needs to be.